tests/ui/match_overlapping_arm.rs

```diff
@@ -57,6 +57,18 @@ fn overlapping() {
         _ => (),
     }
 
+    match 42 {
+        5..7 => println!("5 .. 7"),
+        0..10 => println!("0 .. 10"),
+        _ => (),
+    }
+
+    match 42 {
+        5..10 => println!("5 .. 10"),
+        0..=10 => println!("0 ... 10"),
+        _ => (),
+    }
```

Comment:
Please also add tests for the other two cases, that you mentioned in the PR.

Also can you add a test for 

```
|-----------|    (arm 1)
     |----|      (arm 2)
```

which shouldn't be linted by Clippy, but by rustc.